src/angularComponent/ngcOmniboxSuggestionsDirective.js

```diff
@@ -97,3 +20,62 @@ export default function ngcOmniboxSuggestionsDirective($document, $templateCache
     }
   };
 }
+
+/**
+ * Takes in the public API template submitted by the app-maker and modifies it to repeat through
+ * the data structure of the Omnibox. The template will be different depending on whether it finds
+ * an ngc-omnibox-suggestion-category or not.
+ *
+ * @private
+ * @param {HTMLElement} element -- HTMLElement that contains the template to modify
+ * @param {jQLiteElement} $document
+ * @param {Object} $templateCache
+ * @param {String} templateCacheName -- Unique string to use as the template cache name for
+ *     recursively rendering the template
+ * @returns {String} Modified template string
+ */
+export function getModifiedTemplate(element, $document, $templateCache,
```

Comment:
It seems to me like you're being forced to mix actual params with dependency injections for unit testing in this function now. I think this pattern makes this function worth extracting into a separate file that is an actual Angular service. That way the function itself can only accept the params it needs to perform its actions, rather than mixing dependency injections:

```
myModule.factory('ngcCompileSuggestionsTemplate', ($document, $templateCache) => {
  return function compileSuggestionsTemplate(element) {
    // function body
  };
});
```